### PR TITLE
[fix][broker] Avoid pass null role in MultiRolesTokenAuthorizationProvider

### PR DESCRIPTION
**Pull request by mattisonchao**
### Motivation

Avoid passing the `null` role in `MultiRolesTokenAuthorizationProvider` when the JWT token does not have a corresponding claim.

```
2023-10-31T06:24:46,838+0000 [pulsar-io-4-1] DEBUG org.apache.pulsar.broker.service.ServerCnx - [/10.xx.xx.xx:52096] Authenticate role : admin
2023-10-31T06:24:46,839+0000 [pulsar-io-4-1] DEBUG org.apache.pulsar.broker.service.ServerCnx - [/10.xx.xx.xx:52096] Authenticate original role : d8a70f89-062f-4d9e-9ca5-024d8c5342ec
2023-10-31T06:24:46,839+0000 [pulsar-io-4-1] DEBUG org.apache.pulsar.broker.service.ServerCnx - Authenticate using original auth state : false, role = null
2023-10-31T06:24:46,839+0000 [pulsar-io-4-1] WARN  org.apache.pulsar.broker.authorization.AuthorizationService - [/10.xx.xx.xx:52096] Non-proxy role [admin] passed originalPrincipal [d8a70f89-062f-4d9e-9ca5-024d8c5342ec]. This behavior will not
be allowed in a future release. A proxy's role must be in the broker's proxyRoles configuration.
2023-10-31T06:24:46,839+0000 [pulsar-io-4-1] DEBUG org.apache.pulsar.broker.service.ServerCnx - [/10.xx.xx.xx:52096] connect state change to : [Connected]
2023-10-31T06:24:46,839+0000 [pulsar-io-4-1] DEBUG org.apache.pulsar.broker.service.ServerCnx - [/10.xx.xx.xx:52096] Client successfully authenticated with token role admin and originalPrincipal d8a70f89-062f-4d9e-9ca5-024d8c5342ec
2023-10-31T06:24:46,844+0000 [pulsar-io-4-1] DEBUG org.apache.pulsar.common.protocol.PulsarDecoder - [/10.xx.xx.xx:52096] Received cmd PARTITIONED_METADATA
2023-10-31T06:24:46,844+0000 [pulsar-io-4-1] DEBUG org.apache.pulsar.broker.service.ServerCnx - [persistent://public/default/testabcd] Received PartitionMetadataLookup from /10.xx.xx.xx:52096 for 2833399182903941192
2023-10-31T06:24:46,844+0000 [pulsar-io-4-1] DEBUG org.apache.pulsar.broker.authorization.AuthorizationService - Check if role d8a70f89-062f-4d9e-9ca5-024d8c5342ec is allowed to execute topic operation LOOKUP on topic persistent://public/default/testabcd
2023-10-31T06:24:46,844+0000 [pulsar-io-4-1] DEBUG org.apache.pulsar.broker.authorization.PulsarAuthorizationProvider - Check allowTopicOperationAsync [LOOKUP] on [persistent://public/default/testabcd].
2023-10-31T06:24:46,844+0000 [pulsar-io-4-1] DEBUG org.apache.pulsar.broker.authorization.PulsarAuthorizationProvider - Verify if role null is allowed to LOOKUP to topic persistent://public/default/testabcd: isSuperUserOrAdmin=false
2023-10-31T06:24:46,844+0000 [pulsar-io-4-1] DEBUG org.apache.pulsar.broker.authorization.AuthorizationService - Topic operation LOOKUP on topic persistent://public/default/testabcd is NOT allowed: role = d8a70f89-062f-4d9e-9ca5-024d8c5342ec
2023-10-31T06:24:46,844+0000 [pulsar-io-4-1] DEBUG org.apache.pulsar.broker.authorization.AuthorizationService - Check if role admin is allowed to execute topic operation LOOKUP on topic persistent://public/default/testabcd
2023-10-31T06:24:46,844+0000 [pulsar-io-4-1] DEBUG org.apache.pulsar.broker.authorization.AuthorizationService - Topic operation LOOKUP on topic persistent://public/default/testabcd is allowed: role = admin
2023-10-31T06:24:46,844+0000 [pulsar-io-4-1] WARN  org.apache.pulsar.broker.service.ServerCnx - OriginalRole d8a70f89-062f-4d9e-9ca5-024d8c5342ec is not authorized to perform operation LOOKUP on topic persistent://public/default/testabcd
2023-10-31T06:24:46,844+0000 [pulsar-io-4-1] WARN  org.apache.pulsar.broker.service.ServerCnx - [/10.xx.xx.xx:52096] Client is not authorized to Get Partition Metadata with role d8a70f89-062f-4d9e-9ca5-024d8c5342ec on topic persistent://public/d
```

### Modifications

- Add Null check.
- Add debugging log.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

**Review by Technoboy- — APPROVED** [review]

**Review by RobertIndie — COMMENTED** [review]
LGTM. Could we add a test?

**Review by BewareMyPower — COMMENTED** [review]
+1, it would be better to add a test

**Review by BewareMyPower — APPROVED** [review]

File: pulsar-broker-common/src/test/java/org/apache/pulsar/broker/authorization/MultiRolesTokenAuthorizationProviderTest.java [hunk 1]
```diff
@@ -137,6 +136,41 @@ public String getHttpHeader(String name) {
...
+        };
+
+        assertFalse(provider.authorize("test", ads, role -> {
+            if (role == null) {
+                throw new IllegalStateException("We should avoid pass null to sub providers");
+            }
+            if (role.equals(testRole)) {
+                return CompletableFuture.completedFuture(true);
+            }
+            return CompletableFuture.completedFuture(false);
```
Comment (RobertIndie) [inline pulsar-broker-common/src/test/java/org/apache/pulsar/broker/authorization/MultiRolesTokenAuthorizationProviderTest.java:170]:
```suggestion
            return CompletableFuture.completedFuture(role.equals(testRole));
```

Comment (RobertIndie) [inline pulsar-broker-common/src/test/java/org/apache/pulsar/broker/authorization/MultiRolesTokenAuthorizationProviderTest.java:170]:
```suggestion
            return CompletableFuture.completedFuture(role.equals(testRole));
```

**Review by RobertIndie — APPROVED** [review]

**Pull request merged**